tests/lava/proc/sparse/test_model.py

```diff
@@ -0,0 +1,1451 @@
+        new_weights = np.array([[0, 0], [13, 0]])
+        new_weights_2 = np.array([[32, 0], [13, 0]])
+
+        weights_init_sparse = csr_matrix(weights_init)
+        new_weights_sparse = csr_matrix(new_weights)
+        new_weights_sparse_2 = csr_matrix(new_weights_2)
+
+        inp = (np.random.rand(shape[1], simtime) * 10).astype(int)
+
+        run_cond = RunSteps(num_steps=1)
+        run_cfg = Loihi2SimCfg(select_tag='floating_pt')
+
+        conn = Sparse(weights=weights_init_sparse)
+        sparse_net = create_network(inp, conn, weights_init_sparse)
+        conn.run(condition=run_cond, run_cfg=run_cfg)
+
+        self.assertRaises(ValueError, conn.weights.set, new_weights_sparse)
+        self.assertRaises(ValueError, conn.weights.set, new_weights_sparse_2)
+        conn.stop()
+
+
+class TestLearningSparseProcessModelFloat(unittest.TestCase):
+    """Tests for LearningSparse class in floating point precision. """
+
+    def test_consistency_with_learning_dense_random_shape(self):
+        """Tests if the results of LearningSparse and LearningDense
+        are consistent. """
+
+        simtime = 10
+        shape = np.random.randint(1, 100, 2).tolist()
+        weights = (np.random.random(shape) - 0.5) * 2
+        weights[weights == 0] = 0.1
+
+        learning_rule = STDPLoihi(
+            learning_rate=1,
+            A_plus=1,
+            A_minus=-1,
+            tau_plus=10,
+            tau_minus=10,
+            t_epoch=2,
+        )
+
+        pre = (np.random.rand(shape[1], simtime) > 0.7).astype(int)
+        post = (np.random.rand(shape[0], simtime) > 0.7).astype(int)
+
+        conn = LearningDense(weights=weights,
+                             tag_1=weights.copy(),
+                             tag_2=weights.copy(),
+                             learning_rule=learning_rule)
+        dense_net = create_learning_network(pre, conn, post)
+
+        run_cond = RunSteps(num_steps=simtime)
+        run_cfg = Loihi2SimCfg(select_tag='floating_pt')
+
+        conn.run(condition=run_cond, run_cfg=run_cfg)
+        weights_got_dense = conn.weights.get()
+        conn.stop()
+
+        # Run the same network with Sparse
+
+        # Convert to spmatrix
+        weights_sparse = csr_matrix(weights)
+
+        conn = LearningSparse(weights=weights_sparse,
+                              tag_1=weights_sparse.copy(),
+                              tag_2=weights_sparse.copy(),
+                              learning_rule=learning_rule)
+        sparse_net = create_learning_network(pre, conn, post)
+        conn.run(condition=run_cond, run_cfg=run_cfg)
+
+        weights_got_sparse = conn.weights.get()
+        conn.stop()
+
+        np.testing.assert_array_equal(weights_got_dense,
+                                      weights_got_sparse.toarray())
+
+    def test_consistency_with_learning_dense_random_shape_dt(self):
+        """Tests if the results of LearningSparse and LearningDense
+        are consistent. """
```

Comment:
Clarify in the docstrings of these similar tests how they are different from another.

Reply:
done